src/signed.rs

```diff
@@ -0,0 +1,68 @@
+use crate::{Index, KeyBox};
+use codec::{Decode, Encode};
+
+pub trait Signable {
+    fn bytes_to_sign(&self) -> Vec<u8>;
+}
+
+#[derive(Clone, Debug, Decode, Encode)]
+pub struct UncheckedSigned<T: Signable + Index, S> {
```

Comment:
I would parametrize here by `KB` rather than just by `S`. Even if we don't technically use `KB`, in principle an `UncheckedSigned` is associated with a signing scheme, not just the signature type. And if you follow my suggestion about the `check` method below, it would actually really depend on `KB`.

Hm, after thinking this through before clicking "Finish" – is the problem here that if we parametrize by `KB` we cannot derive all the traits above?

Reply:
We can always implement the traits manually.

Reply:
I tried it, and if we want to parameterize `UncheckedSigned` by the key box, we need to add a ton of boilerplate, which I don't think we should do. I am going it as is